forc-plugins/forc-debug/src/cli/commands.rs

```diff
@@ -1,106 +1,165 @@
-            for c in $names {
-                shell.commands.insert(
-                    c,
-                    ShCommand::new_async($help.to_string(), |state, args| {
-                        Box::pin(async move {
-                            $f(state, args)
-                                .await
-                                .map_err(|e| Box::new(e) as Box<dyn std::error::Error>)
-                        })
-                    }),
-                );
-            }
-        };
+    pub fn is_quit_command(&self, cmd: &str) -> bool {
+        self.quit.name == cmd || self.quit.aliases.contains(&cmd)
+    }
+
+    pub fn is_reset_command(&self, cmd: &str) -> bool {
+        self.reset.name == cmd || self.reset.aliases.contains(&cmd)
     }
 
-    command!(
-        cmd_start_tx,
-        "path/to/tx.json -- start a new transaction",
-        ["n", "tx", "new_tx", "start_tx"]
-    );
-    command!(
-        cmd_reset,
-        "-- reset, removing breakpoints and other state",
-        ["reset"]
-    );
-    command!(
-        cmd_continue,
-        "-- run until next breakpoint or termination",
-        ["c", "continue"]
-    );
-    command!(
-        cmd_step,
-        "[on|off] -- turn single-stepping on or off",
-        ["s", "step"]
-    );
-    command!(
-        cmd_breakpoint,
-        "[contract_id] offset -- set a breakpoint",
-        ["b", "breakpoint"]
-    );
-    command!(
-        cmd_registers,
-        "[regname ...] -- dump registers",
-        ["r", "reg", "register", "registers"]
-    );
-    command!(cmd_memory, "[offset] limit -- dump memory", ["m", "memory"]);
+    pub fn is_continue_command(&self, cmd: &str) -> bool {
+        self.continue_.name == cmd || self.continue_.aliases.contains(&cmd)
+    }
+
+    pub fn is_step_command(&self, cmd: &str) -> bool {
+        self.step.name == cmd || self.step.aliases.contains(&cmd)
+    }
+
+    pub fn is_help_command(&self, cmd: &str) -> bool {
+        self.help.name == cmd || self.help.aliases.contains(&cmd)
+    }
+
+    pub fn find_command(&self, name: &str) -> Option<&Command> {
+        self.all_commands()
+            .into_iter()
+            .find(|cmd| cmd.name == name || cmd.aliases.contains(&name))
+    }
+
+    /// Returns a set of all valid command strings including aliases
+    pub fn get_all_command_strings(&self) -> HashSet<&'static str> {
+        let mut commands = HashSet::new();
+        for cmd in self.all_commands() {
+            commands.insert(cmd.name);
+            commands.extend(cmd.aliases);
+        }
+        commands
+    }
+
+    /// Suggests a similar command
+    pub fn find_closest(&self, unknown_cmd: &str) -> Option<&Command> {
+        self.all_commands()
+            .into_iter()
+            .flat_map(|cmd| {
+                std::iter::once((cmd, cmd.name))
+                    .chain(cmd.aliases.iter().map(move |&alias| (cmd, alias)))
+            })
+            .map(|(cmd, name)| (cmd, levenshtein(unknown_cmd, name)))
```

Comment:
interesting; didnt know there's libs to get closest word :o

Reply:
Yeah it's pretty nifty!